### ISSUE TITLE
A button to turn off notifications from commit messages

### ISSUE THREAD

**Issue by lamont-granquist** [isaacs/github#634]
These are usually incredibly poor submissions.  If a user can't take the time to produce a proper bug report via cutting a new issue, I don't care to have their e-mail in my inbox.  Every single commit that I have is currently a way for a user to send me a nag e-mail and I'd like a way to control that.  When I get e-mail from commit message follow ups its usually something like a user complaining that we've broken an edge case (https://xkcd.com/1172/) -- often because we had to and we explained it in the PR and the release notes -- and so I just have to click 'unsubscribe' on the commit message so that i don't hear it any more.  I'd like to just ignore them completely.  I'm also fine with the ability to lock merged commits so that nobody can comment.  The current behavior is that I wind up essentially shadowbanning people.  There may be some commit message of mine out there with 800 replies to it demanding that I revert it and I'll never know about it (although that might be a good thing).  If it was just locked instead then it would force users with complaints to the actual issue tracker for the project where the issue could go through our triage process and get addressed one way or the other.

**Comment by cirosantilli** [comment]
Do you mean comments on commits, e.g. https://github.com/chef/chef/commit/678101bdf560789d2408ce44720732035db978c3 at the bottom of the page? If so, I recommend changing the title to just "A button to turn off notifications from commit" Otherwise people will think about at mentions on commit messages.

It is always a good idea to give a sample URL that demonstrates the issue ;-)

**Comment by lamont-granquist** [comment]
I don't see how your title is any better than mine.  I didn't say anything about at-mentions in the title or in the body, so its very confusing to me as to why you'd think I was talking about at-mentions.  Anyway its not about at-mentions, its about unsubscribing to the 'threads' that seem to be created by individual commits.

**Comment by cirosantilli** [comment]
Feel free to reverse the title.

**Comment by cirosantilli** [comment]
Reversed the title.

**Comment by nmccready** [comment]
ref: #38

**Comment by lamont-granquist** [comment]
yeah the issues in #38 are close, i more want to just unsubscribe and ignore someone troll baiting me in a comment.  being able to lock a commit thread would be an additional feature.   i'd also like the ability to lock all my commit threads from commits that have been merged in PRs.  ideally i'd really want to force someone not to give me a drive-by in some ancient commit, but force them to open a new issue if they really have a new issue.

**Comment by lamont-granquist** [comment]
and indeed the comment thread in f51b931c831ed405e535025aea24dafdffb11e11 is an excellent example of exactly what i'm talking about...

**Comment by nmccready** [comment]
Yes I ref 38 because in my mind it is not complete as this request is a must. I did not ref 38 to be a troll.
